client/reader/sidebar/expandable.jsx

```diff
@@ -37,11 +37,11 @@ const ExpandableSidebarMenu = React.createClass( {
 				'is-togglable': true
 			}
 		);
-
+		const shouldHideAddButton = ! this.props.count && this.props.title === 'Lists';
```

Comment:
I'd lift this out to `reader/sidebar/reader-sidebar-lists/index` and pass it through to `Expandable` via a prop. And then just default the prop to true.